tests/integration/test_data/npm_smoketest_lockfile2/bom.json

```diff
@@ -0,0 +1,371 @@
+{
+  "bomFormat": "CycloneDX",
+  "components": [
+    {
+      "name": "bar",
+      "version": "1.0.0",
+      "type": "library"
+    },
+    {
+      "name": "baz",
+      "version": "1.0.0",
+      "type": "library"
+    },
+    {
+      "name": "eggs",
+      "version": "1.0.0",
+      "type": "library"
+    },
+    {
+      "name": "foo",
+      "version": "1.0.0",
+      "type": "library"
+    },
+    {
+      "name": "abbrev",
+      "version": "2.0.0",
+      "type": "library"
+    },
```

Comment:
The BOMs are not sorted, which makes them difficult to `diff` (and during implementation, they will become sorted) 

I sorted them locally like this
```bash
sort_bom() {
    content=$(cat "$1")
    jq <<< "$content" '.components = (.components | sort_by(.version) | sort_by(.name))' > "$1"
}

sort_bom tests/integration/test_data/npm_smoketest_lockfile1/bom.json
sort_bom tests/integration/test_data/npm_smoketest_lockfile2/bom.json
sort_bom tests/integration/test_data/npm_smoketest_lockfile3/bom.json
```

Reply:
This is the sbom I'm getting from cachi2 for the v2 lockfile, if it helps with the sort order: https://privatebin.corp.redhat.com/?debecb2056d27518#7zK69MyXFNKBZhup6EUsxz77B2RaDFhspHfjLbL7ADGM

Cachi2 components look like they're being sorted with the key "name:version"

Reply:
Ah right, amending my `sort_bom`:

```shell
jq <<< "$content" '.components | sort_by("\(.name):\(.version // "None")")'
```

Side note, this is why shouldn't sort by `name:version` but by `(name, version)` (but the sorting key is supposed to be temporary anyway)

```python

In [1]: ("cookie", "0.5.0") < ("cookie-signature", "1.0.6")
Out[1]: True

In [2]: ("cookie:0.5.0") < ("cookie-signature:1.0.6")
Out[2]: False
```

Reply:
Nice sorting method! :taco: